api/v1beta1/awsclusterstaticidentity_webhook.go

```diff
@@ -65,16 +65,11 @@ func (r *AWSClusterStaticIdentity) ValidateDelete() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
 func (r *AWSClusterStaticIdentity) ValidateUpdate(old runtime.Object) error {
-	oldP, ok := old.(*AWSClusterStaticIdentity)
+	_, ok := old.(*AWSClusterStaticIdentity)
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected an AWSClusterStaticIdentity but got a %T", old))
 	}
 
-	if oldP.Spec.SecretRef != r.Spec.SecretRef {
```

Comment:
in update case, if we change the identity, this check will fail

Reply:
I might be missing something but this checks if there is any change on the static identity itself. How does this block changing identity in AWSCluster?

Reply:
oldP.Spec.SecretRef will point to the old identity secretRef, but after update case, r.Spec.SecretRef will point to the new identity, hence these two wont match and it will fail this check.